contracts/account/modules/ERC7579DelayedExecutor.sol

```diff
@@ -0,0 +1,458 @@
+     * override this to implement custom authorization logic.
+     *
+     * Example extension:
+     *
+     * ```solidity
+     *  function _validateCancel(
+     *     address account,
+     *     bytes32 mode,
+     *     bytes calldata data,
+     *     bytes32 salt
+     *  ) internal view override returns (bool) {
+     *    bool isAuthorized = ...; // custom logic to check authorization
+     *    return isAuthorized || super._validateCancel(account, mode, data, salt);
+     *  }
+     *```
+     */
+    function _validateCancel(
+        address account,
+        bytes32 /* salt */,
+        bytes32 /* mode */,
+        bytes calldata /* data */
+    ) internal view virtual returns (bool) {
+        return account == msg.sender;
+    }
+
+    /**
+     * @dev Whether the caller is authorized to schedule operations.
+     * By default, this checks if the caller is the account itself. Derived contracts can
+     * override this to implement custom authorization logic.
+     *
+     * Example extension:
+     *
+     * ```solidity
+     *  function _validateSchedule(
+     *     address account,
+     *     bytes32 mode,
+     *     bytes calldata data,
+     *     bytes32 salt
+     *  ) internal view override returns (bool) {
+     *    bool isAuthorized = ...; // custom logic to check authorization
+     *    return isAuthorized || super._validateSchedule(account, mode, data, salt);
+     *  }
+     *```
+     */
+    function _validateSchedule(
+        address account,
+        bytes32 /* salt */,
+        bytes32 /* mode */,
+        bytes calldata /* data */
+    ) internal view virtual returns (bool) {
+        return account == msg.sender;
+    }
+
+    /**
+     * @dev Internal implementation for setting an account's delay. See {getDelay}.
+     *
+     * Emits an {ERC7579ExecutorDelayUpdated} event.
+     */
+    function _setDelay(address account, uint32 newDelay, uint32 minimumSetback) internal virtual {
+        uint48 effect;
+        (_config[account].delay, effect) = _config[account].delay.withUpdate(newDelay, minimumSetback);
+        emit ERC7579ExecutorDelayUpdated(account, newDelay, effect);
+    }
+
+    /**
+     * @dev Internal implementation for setting an account's expiration. See {getExpiration}.
+     *
+     * Emits an {ERC7579ExecutorExpirationUpdated} event.
+     */
+    function _setExpiration(address account, uint32 newExpiration) internal virtual {
+        // Safe downcast since both arguments are uint32
+        _config[account].expiration = newExpiration;
+        emit ERC7579ExecutorExpirationUpdated(account, newExpiration);
+    }
+
+    /**
+     * @dev Internal version of {schedule} that takes an `account` address as an argument.
+     *
+     * Requirements:
+     *
+     * * The operation must be `Unknown`.
+     *
+     * Emits an {ERC7579ExecutorOperationScheduled} event.
+     */
+    function _schedule(
```

Comment:
I'm thinking this one should instead be `_scheduleAt`. So that the public `schedule` uses `_scheduleAt` and passes in the delay. The idea is that developers would have an internal function that allows them to schedule at specific timepoints if needed. The current `_schedule` does not allow for that.

It should have a note, though, making it clear that exposing the function might allow bypassing the regular delay